crates/nu-parser/src/deparse.rs

```diff
@@ -1,14 +1,25 @@
 pub fn escape_quote_string(input: &str) -> String {
```

Comment:
I think we can extract `for loop` into a function to avoid duplicate code (just a new one's perspective, don't konw how it works internally)?  Like this:

```rust
fn escape_input(input: &str, output: &mut String) {
    for c in input.chars() {
        if c == '"' || c == '\\' {
            output.push('\\');
        }
        output.push(c);
    }
}

pub fn escape_quote_string(input: &str) -> String {
    if input.starts_with('-') {
        let mut output = String::with_capacity(input.len());
        escape_input(input, &mut output);
        output
    } else {
        let mut output = String::with_capacity(input.len() + 2);
        output.push('"');
        escape_input(input, &mut output);
        output.push('"');
        output
    }
}
```

And also, if input is starts with '-', we can `let mut output = String::with_capacity(input.len())`.  Because we can make sure that output has at least input length.

Reply:
good suggestions, thanks!